File: problem-solving/parsers/py/_10.py
def parser(f):
	n = int(f.readline())
	ticket = f.readline().strip()

	assert(n == len(ticket))
	for c in ticket:
		assert(c == '*' or ('0' <= c and c <= '9'))
	assert(f.readline() == "\n")

	cert = f.readline().strip()
	assert(cert == "true" or cert == "false")
	assert(f.readline().strip() == "")
	if cert == "true":
		cert = True
	else:
		cert = False

	return cert, (n, ticket,)

def verifier(cert, ans):
	return cert == ans

def error(cert, input, ans):
	print("Input:")
	print("n = " + str(input[0]))
	print("ticket = " + str(input[1]))
	print("")
	print("Expected:")
	print(cert)
	print("")
	print("Actual:")
	print(ans)
	print()

File: problem-solving/parsers/java/_10.java
import java.util.Scanner;

public class _10 implements Parser {
    public Object[] parser(Scanner f) {
        int n = f.nextInt();
        f.nextLine();
        String ticket = f.nextLine();

        assert(f.nextLine() == "");

        boolean cert = f.nextBoolean();

        return Parser.ret_parser(cert, n, ticket);
    }

    public Object run(Object[] input) {
        return Solutions.reviveStrings((int)input[0], (String)input[1]);
    }

    public boolean verifier(Object cert, Object ans) {
        return (boolean)cert == (boolean)ans;
    }

    public void error(Object cert, Object[] input, Object ans) {
        System.out.println(String.format(
            "Input:\n" +
            "n = %d\n" +
            "ticket = %s\n" +
            "\n" +
            "Expected:\n" +
            "%b\n" +
            "\n" +
            "Actual:\n" +
            "%b\n",
            (int)input[0],
            (String)input[1],
            (boolean)cert,
            (boolean)ans
        ));
    }

}
